File: Frontend/scripts/services/validAccount.js
const BASE_URL = "http://127.0.0.1:8000";

async function validAccount(typeAccount, mail, password) {
    const query = BASE_URL + "/valid/?" + "typeAccount=" + typeAccount + "&mail=" + mail + "&password=" + password;

    const response = await fetch(query);

    if(response.status === 404) {
        throw new Error("No encontrado");
    }

    if(!response.ok) {
        throw new Error("Error HTTP" + response.status);
    }

    const data = await response.json();

    return data
}


// --- EXPORT FUNCTION ---

export async function validAccountService(typeAccount, mail, password) {
    return await validAccount(typeAccount, mail, password)    
}

File: Backend/src/services/validAccount.py
from fastapi import APIRouter, HTTPException
from models.tutor import Tutor
from models.school import School

router = APIRouter()

@router.get("/")
def validAccount(typeAccount: str, mail: str, password: str):
    
    if(typeAccount != "school" and typeAccount != "tutor"):
        raise HTTPException(status_code=404, detail="Tipo de cuenta invalido.")
    
    if(typeAccount == "school"):
        temp = School.get_by("mail", mail)
        
        if not temp["ok"]:
            raise HTTPException(status_code=500, detail="Error interno")
        
        if not temp["data"]:
            return {"ok": False, "data": "isNotValid"}
        
        passTemp = temp["data"][0][3]
        
        if(password == passTemp):
            return {"ok": True, "data": "isValid"}
        else:
            return {"ok": False, "data": "isNotValid"}

    if(typeAccount == "tutor"):
        temp = Tutor.get_by("mail", mail)
        
        if not temp["ok"]:
            raise HTTPException(status_code=500, detail="Error interno")
        
        if not temp["data"]:
            return {"ok": False, "data": "isNotValid"}
        
        passTemp = temp["data"][0][4]
        
        if(password == passTemp):
            return {"ok": True, "data": "isValid"}
        else:
            return {"ok": False, "data": "isNotValid"}
        
    return {"ok": False, "data": "Error"}